demo/src/main/java/com/example/demo/services/impl/DrinkServiceImpl.java

```diff
@@ -1,36 +1,63 @@
 package com.example.demo.services.impl;
 
+import com.example.demo.dto.drink.DrinkInfoDto;
 import com.example.demo.entities.Drink;
 import com.example.demo.repository.DrinkRepository;
 import com.example.demo.services.DrinkService;
+import org.springframework.transaction.annotation.Transactional;
 import lombok.RequiredArgsConstructor;
 import org.springframework.stereotype.Service;
 
 import java.util.List;
+import java.util.stream.Collectors;
 
 @Service
 @RequiredArgsConstructor
 public class DrinkServiceImpl implements DrinkService {
 
     private final DrinkRepository drinkRepository;
-    @Override
-    public Drink create(Drink drink) {
+
+    private Drink mapToDrink(DrinkInfoDto drinkInfoDto){
+        Drink drink = new Drink();
+        drink.setName(drinkInfoDto.getName());
+        drink.setPrice(drinkInfoDto.getPrice());
+        drink.setPresence(drinkInfoDto.getPresence());
         return drinkRepository.save(drink);
     }
+
+    private DrinkInfoDto mapToDrinkDto(Drink drink){
+        DrinkInfoDto dto = new DrinkInfoDto();
+        dto.setName(drink.getName());
+        dto.setPrice(drink.getPrice());
+        dto.setPresence(drink.getPresence());
+        return dto;
+    }
+
     @Override
-    public Drink read(Long id) {
-        return drinkRepository.findById(id).orElse(null);
+    @Transactional
+    public DrinkInfoDto create(DrinkInfoDto createDrinkDTO) {
+       if(createDrinkDTO != null){
+           Drink savedDrink = mapToDrink(createDrinkDTO);
+           return mapToDrinkDto(savedDrink);
+       }
+       return null;
```

Comment:
This is a really bad practice to return nulls on persistent entity creation. User will not expect nulls in return values and you will breake any system which will use your API later. In case your method arguments expect some data but not get it -> throw an error with a description.